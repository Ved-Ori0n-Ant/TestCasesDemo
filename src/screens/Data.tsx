export const Data = [
    {
      id: 1,
      name: 'Ivy NextShares',
    },
    {
      id: 2,
      name: 'Ottawa Bancorp, Inc.',
    },
    {
      id: 3,
      name: 'Alder BioPharmaceuticals, Inc.',
    },
    {
      id: 4,
      name: 'Principal Healthcare Innovators Index ETF',
    },
    {
      id: 5,
      name: 'Canada Goose Holdings Inc.',
    },
    {
      id: 6,
      name: 'Aralez Pharmaceuticals Inc.',
    },
    {
      id: 7,
      name: 'China Mobile (Hong Kong) Ltd.',
    },
    {
      id: 8,
      name: 'J P Morgan Chase & Co',
    },
    {
      id: 9,
      name: 'Matlin & Partners Acquisition Corporation',
    },
    {
      id: 10,
      name: 'Southern California Edison Company',
    },
    {
      id: 11,
      name: 'Live Nation Entertainment, Inc.',
    },
    {
      id: 12,
      name: 'Wells Fargo & Company',
    },
    {
      id: 13,
      name: 'TransAct Technologies Incorporated',
    },
    {
      id: 14,
      name: 'YY Inc.',
    },
    {
      id: 15,
      name: 'Entertainment Gaming Asia Incorporated',
    },
    {
      id: 16,
      name: 'Two River Bancorp',
    },
    {
      id: 17,
      name: 'MEI Pharma, Inc.',
    },
    {
      id: 18,
      name: 'Cerecor Inc.',
    },
    {
      id: 19,
      name: 'Petrobras Argentina S.A.',
    },
    {
      id: 20,
      name: 'PowerShares DWA Momentum Portfolio',
    },
    {
      id: 21,
      name: 'iShares S&P Global Timber & Forestry Index Fund',
    },
    {
      id: 22,
      name: 'Boot Barn Holdings, Inc.',
    },
    {
      id: 23,
      name: 'Natural Health Trends Corp.',
    },
    {
      id: 24,
      name: 'VictoryShares Emerging Market Volatility Wtd ETF',
    },
    {
      id: 25,
      name: 'Engility Holdings, Inc.',
    },
    {
      id: 26,
      name: 'CECO Environmental Corp.',
    },
    {
      id: 27,
      name: 'Micronet Enertec Technologies, Inc.',
    },
    {
      id: 28,
      name: 'Validea Market Legends ETF',
    },
    {
      id: 29,
      name: 'Alphatec Holdings, Inc.',
    },
    {
      id: 30,
      name: 'Capitala Finance Corp.',
    },
  ];