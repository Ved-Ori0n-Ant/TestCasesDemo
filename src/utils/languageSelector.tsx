import React from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Dimensions,
  Image,
} from 'react-native';
import {useTranslation} from 'react-i18next';
import '../localization';

const {width} = Dimensions.get('window');

const LANGS = [
  {lngCode: 'en', label: 'English'},
  {lngCode: 'hi', label: 'हिन्दी'},
  {lngCode: 'ar', label: 'عربي'},
];

const LanguageSelector = () => {
  const {t, i18n} = useTranslation();
  const selectedLngCode: any = i18n.language;
  const setLng = (lngCode: string) => i18n.changeLanguage(lngCode);
  return (
    <View style={styles.container}>
      <View style={styles.titleContainer}>
        {/* Should return select your language in respective language */}
        <Text style={styles.select}>{t('languageSelector:selectLng')}</Text> 
      </View>
      {LANGS.map((l) => {
        const selected = l.lngCode === selectedLngCode;
        return (
          <TouchableOpacity
            onPress={() => setLng(l.lngCode)}
            key={l.lngCode}
            disabled={selected}>
            <View style={[styles.row, selected ? styles.selectedRow : {}]}>
              <Text style={[selected ? styles.selectedText : styles.text]}>
                {l.label}
              </Text>
              {selected && <Image source={require('../utils/images/world.png')} style={{height: 20, width:20}} />}
            </View>
          </TouchableOpacity>
        );
      })}
    </View>
  );
};

export default LanguageSelector;

const styles = StyleSheet.create({
  container: {
    padding: 10,
    backgroundColor: 'rgb(231, 232, 235)',
    width: width * 0.8,
  },
  titleContainer: {
    alignItems: 'center',
    paddingBottom: 20,
  },
  select: {
    fontSize: 20,
  },
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: 10,
    paddingVertical: 5,
  },
  selectedRow: {
    backgroundColor: 'rgb(45, 45, 45)',
  },
  selectedText: {
    color: 'rgb(231, 232, 235)',
  },
  text: {
    color: 'rgb(45, 45, 45)',
  },
});