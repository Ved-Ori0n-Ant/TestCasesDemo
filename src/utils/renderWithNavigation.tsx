import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {render} from '@testing-library/react-native';

export function renderWithNavigation(renderComponent: any) {
  return render(<NavigationContainer>{renderComponent}</NavigationContainer>);
}
