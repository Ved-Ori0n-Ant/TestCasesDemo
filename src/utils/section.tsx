import React, {ReactNode} from 'react';
import {StyleSheet, View, Text} from 'react-native';

interface Props {
  children: ReactNode;
  title: string;
  testID?: string;
}

const Section = ({title, children, testID}: Props) => {
  return (
    <View style={styles.container} testID={testID}>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{title}</Text>
      </View>
      {children}
    </View>
  );
};

export default Section;

const styles = StyleSheet.create({
  container: {
    alignContent: 'center',
    marginVertical: 10,
    padding: 10,
    justifyContent: 'center',
    width: '80%',
  },
  titleContainer: {
    alignItems: 'center',
  },
  title: {
    fontSize: 26,
    fontWeight: 'bold',
    color: 'maroon',
  },
});