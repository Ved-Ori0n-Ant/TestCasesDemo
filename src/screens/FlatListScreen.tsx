import React from 'react';
import {
  FlatList,
  SafeAreaView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import NextnPrevious, {Stack} from '../components/NextnPrevious';
import { Data } from './Data'

const FlatListScreen = () => {
  const ListItems = ({
    value,
    index,
  }: {
    value: {name: string; id: number};
    index: number;
  }) => (
    <TouchableOpacity
      onPress={() => {
        console.log(value.name);
      }}
      key={index.toString()}
      testID={'FlatList-Item'}
      style={{
        paddingVertical: 16,
        padding: 8,
        borderRadius: 12,
        backgroundColor: '#fff',
        marginHorizontal: 16,
        marginBottom: 16,
        justifyContent: 'center',
        alignItems: 'center',
      }}>
      <Text testID={'FlatList-ItemName'}>{value.name}</Text>
    </TouchableOpacity>
  );
  return (
    <SafeAreaView style={styles.body}>
      <View style={styles.subBody}>
        <FlatList
          testID={'FlatList'}
          data={Data}
          renderItem={({item, index}) => (
            <ListItems value={item} index={index} />
          )}
        />
      </View>
      <NextnPrevious currentScreen={Stack.FlatListScreen} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  body: {
    flex: 1,
  },
  subBody: {
    flex: 1,
    backgroundColor: '#ddd',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default FlatListScreen;
