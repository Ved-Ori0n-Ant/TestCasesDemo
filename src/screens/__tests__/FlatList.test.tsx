import { renderWithNavigation } from "../../utils/renderWithNavigation";
import { Data } from "../Data";
import FlatListScreen from "../FlatListScreen";
import React from 'react';

describe('Tests run on FlatList', () => {
    it('should get rendered', () => {
        const tree = renderWithNavigation(<FlatListScreen />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
    it('should have FlatList', () => {
        const tree = renderWithNavigation(<FlatListScreen />);
        const flatListItemList = tree.getByTestId('FlatList').props
        expect(flatListItemList).not.toBeNull();
        expect(flatListItemList.data.length).toBe(30)
    });
    it('should have itemList rendered', () => {
        const tree = renderWithNavigation(<FlatListScreen />);
        const item = tree.getAllByTestId('FlatList-ItemName');

        // Scrolled is not applied thus only 10 items will be shown
        expect(item.length).toBe(10)
        item.forEach((value, index) => {
            // console.log(value.props.children, '==?==', Data[index].name)
            expect(value.props.children).toBe(Data[index].name)
        });
    })
})