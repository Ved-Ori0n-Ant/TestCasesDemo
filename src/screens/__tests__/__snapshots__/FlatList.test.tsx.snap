// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests run on FlatList should get rendered 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ddd",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "id": 1,
            "name": "Ivy NextShares",
          },
          {
            "id": 2,
            "name": "Ottawa Bancorp, Inc.",
          },
          {
            "id": 3,
            "name": "Alder BioPharmaceuticals, Inc.",
          },
          {
            "id": 4,
            "name": "Principal Healthcare Innovators Index ETF",
          },
          {
            "id": 5,
            "name": "Canada Goose Holdings Inc.",
          },
          {
            "id": 6,
            "name": "Aralez Pharmaceuticals Inc.",
          },
          {
            "id": 7,
            "name": "China Mobile (Hong Kong) Ltd.",
          },
          {
            "id": 8,
            "name": "J P Morgan Chase & Co",
          },
          {
            "id": 9,
            "name": "Matlin & Partners Acquisition Corporation",
          },
          {
            "id": 10,
            "name": "Southern California Edison Company",
          },
          {
            "id": 11,
            "name": "Live Nation Entertainment, Inc.",
          },
          {
            "id": 12,
            "name": "Wells Fargo & Company",
          },
          {
            "id": 13,
            "name": "TransAct Technologies Incorporated",
          },
          {
            "id": 14,
            "name": "YY Inc.",
          },
          {
            "id": 15,
            "name": "Entertainment Gaming Asia Incorporated",
          },
          {
            "id": 16,
            "name": "Two River Bancorp",
          },
          {
            "id": 17,
            "name": "MEI Pharma, Inc.",
          },
          {
            "id": 18,
            "name": "Cerecor Inc.",
          },
          {
            "id": 19,
            "name": "Petrobras Argentina S.A.",
          },
          {
            "id": 20,
            "name": "PowerShares DWA Momentum Portfolio",
          },
          {
            "id": 21,
            "name": "iShares S&P Global Timber & Forestry Index Fund",
          },
          {
            "id": 22,
            "name": "Boot Barn Holdings, Inc.",
          },
          {
            "id": 23,
            "name": "Natural Health Trends Corp.",
          },
          {
            "id": 24,
            "name": "VictoryShares Emerging Market Volatility Wtd ETF",
          },
          {
            "id": 25,
            "name": "Engility Holdings, Inc.",
          },
          {
            "id": 26,
            "name": "CECO Environmental Corp.",
          },
          {
            "id": 27,
            "name": "Micronet Enertec Technologies, Inc.",
          },
          {
            "id": 28,
            "name": "Validea Market Legends ETF",
          },
          {
            "id": 29,
            "name": "Alphatec Holdings, Inc.",
          },
          {
            "id": 30,
            "name": "Capitala Finance Corp.",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      testID="FlatList"
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Ivy NextShares
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Ottawa Bancorp, Inc.
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Alder BioPharmaceuticals, Inc.
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Principal Healthcare Innovators Index ETF
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Canada Goose Holdings Inc.
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Aralez Pharmaceuticals Inc.
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              China Mobile (Hong Kong) Ltd.
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              J P Morgan Chase & Co
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Matlin & Partners Acquisition Corporation
            </Text>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 12,
                "justifyContent": "center",
                "marginBottom": 16,
                "marginHorizontal": 16,
                "opacity": 1,
                "padding": 8,
                "paddingVertical": 16,
              }
            }
            testID="FlatList-Item"
          >
            <Text
              testID="FlatList-ItemName"
            >
              Southern California Edison Company
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="np_btn"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Previous_btn"
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Previous
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="Next_btn"
    >
      <View
        style={
          [
            {},
            {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
